components/Buttons/src/MDCButton.m

```diff
@@ -727,11 +724,7 @@ - (void)setBorderColor:(UIColor *)borderColor forState:(UIControlState)state {
 #pragma mark - Border Width
 
 - (CGFloat)borderWidthForState:(UIControlState)state {
-  NSNumber *borderWidth = _borderWidths[@(state)];
-  if (borderWidth) {
-    return (CGFloat)borderWidth.doubleValue;
-  }
-  return 0;
+  return (CGFloat)_borderWidths[@(state)].doubleValue;
```

Comment:
The implementation of borderWidthForState: and elevationForState: are different.
elevation returns valueForState:UIControlStateNormal if valueForState:state doesn't exist.
This one returns 0 and handles in updateBorderWidth.

Please migrate borderWidthForState to the same implementation as elevationForState.

Understood if you want to do this in a different PR.

Reply:
Moved logic from `updateBorderWidth`.

Should we apply the same logic for `titleFontForState:`?